src/material/schematics/ng-generate/theming-api/migration.ts

```diff
@@ -0,0 +1,251 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/** Mapping of Material mixins that should be renamed. */
+const materialMixins: Record<string, string> = {
+  'mat-core': 'core',
+  'mat-core-color': 'core-color',
+  'mat-core-theme': 'core-theme',
+  'angular-material-theme': 'all-component-themes',
+  'angular-material-typography': 'all-component-typographies',
+  'angular-material-color': 'all-component-colors',
+  'mat-base-typography': 'typography-hierarchy',
+  'mat-typography-level-to-styles': 'typography-level',
+  'mat-elevation': 'elevation',
+  'mat-overridable-elevation': 'overridable-elevation',
+  'mat-ripple': 'ripple',
+  'mat-ripple-color': 'ripple-color',
+  'mat-ripple-theme': 'ripple-theme',
+  'mat-strong-focus-indicators': 'strong-focus-indicators',
+  'mat-strong-focus-indicators-color': 'strong-focus-indicators-color',
+  'mat-strong-focus-indicators-theme': 'strong-focus-indicators-theme',
+  'mat-font-shorthand': 'font-shorthand',
+  // The expansion panel is a special case, because the package is called `expansion`, but the
+  // mixins were prefixed with `expansion-panel`. This was corrected by the Sass module migration.
+  'mat-expansion-panel-theme': 'expansion-theme',
+  'mat-expansion-panel-color': 'expansion-color',
+  'mat-expansion-panel-typography': 'expansion-typography',
```

Comment:
@jelbourn as I was compiling these mappings, I realized that the new file doesn't expose some symbols that we had in the old theming.scss (e.g. `$mat-small`). I feel like they might've been exposed by accident due to the old Sass bundler. Should we re-expose them for backwards-compatibility?

Reply:
I think we should be okay omitting them, but we should document the omissions in the changelog as breaking changes. The explanation would be that these APIs were meant for internal use only but were accidentally exposed.